ui.apps/src/main/content/jcr_root/apps/core/cif/clientlibs/storefront-events/js/collector.js

```diff
@@ -17,3 +17,18 @@ import mse from '@adobe/magento-storefront-events-sdk';
 
 // Expose Magento Storefront Events SDK on the global window object
 window.magentoStorefrontEvents = mse;
+
+const processProductStorefrontData = () => {
```

Comment:
We should wrap this part into an IIFE and also check for the DOM to be ready. Similar to https://github.com/adobe/aem-core-wcm-components/blob/development/content/src/scripts/datalayer/v1/datalayer.js#L100-L104.